### DISCUSSION TITLE
ARTPEC-8 CPU slow execution speed

### DISCUSSION THREAD

**Discussion by jklontz** [AxisCommunications/discussions#1138] [ACAP] [answered]
We're benchmarking our native computer vision analytics application on a P3268-LVE by copying it onto the camera over ssh and executing it from the shell. From open source SPECint benchmarks for Cortex A53 CPUs we were expecting to measure approximately a 20x slowdown in single-core execution speed relative to our baseline desktop-class processor. In practice we're measuring closer to a 90x slowdown. We've confirmed the usual suspects (Release build flags, NEON vectorization) are correctly enabled. Our two questions are:
    1. What is the clock speed of the ARTPEC-8?
    2. Does the OS throttle processes running via the shell, or should we expect to not have a full CPU core available?

**Comment by github-actions** [comment]
*This automatically generated reply acts as a friendly reminder.*

*Answers to your questions will most often come from the community, from developers like yourself. You will, from time to time, find that Axis employees answers some of the questions, but this is not a guarantee. Think of the discussion forum as a complement to other support channels, not a replacement to any of them.*

*If your question remains unanswered for a period of time, please revisit it to see whether it can be improved by following the guidelines listed in [Axis support guidelines](https://github.com/orgs/AxisCommunications/discussions/13).*

**Comment by vivekatoffice** [comment, marked as answer]
Hi @jklontz ,
Why not use the **"axis-a8-dlpu-tflite"**?

axis-XXXXXXXXX:~$ larod-client -a
2025-09-11T09:48:27.507 Connecting to larod...
2025-09-11T09:48:27.875 Connected
2025-09-11T09:48:27.875 Devices:
___________________________________________
 Name                             Instance
‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
 cpu-tflite                       0
 axis-ace-proc                    0
 cpu-proc                         0
 axis-a8-dlpu-tflite              0
 axis-a8-dlpu-native              0
 axis-a8-dlpu-proc                0
 axis-a8-gpu-proc                 0
___________________________________________

1. What is the clock speed of the ARTPEC-8?

> 4 CPUs 1200000 each (1200000 kHz = 1200 MHz = 1.2 GHz)
<img width="785" height="247" alt="image" src="https://github.com/user-attachments/assets/33f659e1-c01d-4422-9056-e09b317f6671" />

```bash
axis-XXXXXXXXX:~$ cat /sys/devices/system/cpu/cpu3/cpufreq/cpuinfo_max_freq
1200000
axis-XXXXXXXXX:~$ cat /sys/devices/system/cpu/cpu2/cpufreq/cpuinfo_max_freq
1200000
axis-XXXXXXXXX:~$ cat /sys/devices/system/cpu/cpu1/cpufreq/cpuinfo_max_freq
1200000
axis-XXXXXXXXX:~$ cat /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq
1200000
```

3. Does the OS throttle processes running via the shell, or should we expect not to have a full CPU core available?

> Do you have access to the [Integration Helpdesk](https://www.axis.com/support/helpdesk)? You can ask this question here as an [Axis Technology Integration Partner Program](https://www.axis.com/partner/technology-integration-partner-program) member 😉 

[Integration Helpdesk](https://www.axis.com/support/helpdesk)
<img width="1035" height="539" alt="image" src="https://github.com/user-attachments/assets/13b0abe1-17f4-45ee-ad93-9dfeadf7c3f1" />


### Edit: According to one of the ACAP experts:

#### There is no throttling of processes running via shell. Developers can use the shell command 'top' to see the CPU usage of their application via SSH.

<img width="918" height="695" alt="image" src="https://github.com/user-attachments/assets/f99c4fde-51a0-4026-9fe0-d501c2da54fc" />

#### If partner developers are using TFLite, I suspect the performance difference is due to different versions of TensorFlow.

**Are you using Tensorflow for your native computer vision analytics application? Version of Tensorflow?**

**Comment by jklontz** [comment]
Thanks for the robust answers here. Would agree that the DLPU is the only viable path forward here. We have some reservations about the challenges converting Pytorch ==> ONNX ==> TFlite but should ultimately be surmountable with some trial and error...
